scripts/config-plugins/plugins/mod-compiler.mjs

```diff
@@ -13,6 +14,12 @@ export const withDefaultBaseMods = (config, props) => {
     ...props,
     providers: getAndroidModFileProviders(),
   });
+  config = BaseMods.withGeneratedBaseMods(config, {
+    ...props,
+    // @ts-expect-error `macos` is not assignable to type `android | ios`
+    platform: "macos",
```

Comment:
Unfortunately, Expo config plugins hard-code the string "ios" all over the place, so even if you pass in the string `macos`, it'll apply changes under the `ios` folder, e.g. to `ios/Example/Info.plist`.

I opened a big [PR](https://github.com/expo/expo/pull/30660) last weekend to add macOS support to Expo config plugins, but I am not optimistic about it being reviewed anytime soon.

Reply:
If I understand this correctly, this part just registers what's available for the `macos` platform. We need to explicitly implement the mods separately, using `withMod`, as in [`react-native-test-app/plugins/index.js`](https://github.com/microsoft/react-native-test-app/blob/90ae7901ce523404faa31a62bf0b58a133d4125f/plugins/index.js#L15-L55).